Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1097 @@
+#include "WarpX.H"
+#include <AMReX_Scan.H>
+#include <AMReX_iMultiFab.H>
+#include <AMReX_MultiFab.H>
+
+/**
+* \brief auxiliary function to count the amount of faces which still need to be extended
+*/
+amrex::Array1D<int, 0, 2>
+WarpX::CountExtFaces() {
+    amrex::Array1D<int, 0, 2> sums{0, 0, 0};
+#ifdef AMREX_USE_EB
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        int sum = 0;
+        for (amrex::MFIter mfi(*m_flag_ext_face[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+            amrex::Box const &box = mfi.validbox();
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            amrex::ParallelFor(box, [=, &sum] AMREX_GPU_DEVICE (int i, int j, int k) {
```

Comment:
@WeiqunZhang is there a better way of doing this? I am essentially trying to count the number of `true` entries in `flag_ext_face`

Reply:
This does not work on gpu.  You can take a look at this https://github.com/ECP-WarpX/WarpX/blob/development/Source/Diagnostics/ReducedDiags/FieldMomentum.cpp#L142 for an example of doing 3 sums.  Do you have access to a machine where you can test this PR on GPU?

Reply:
I think I fixed it, thanks a lot for the suggestion. It would be amazing if you could have a look at the new code. Unfortunately, I don't have access to a machine with a GPU right now, but I'm working on it.